### PR TITLE
ref #689: cronjobs construct fix

### PR DESCRIPTION
**Pull request by kzorluoglu**
| Q             | A
| ------------- | ---
| Branch        | 7.1.x for bug fixes 
| Bug fix?      | yes
| New feature?  | 
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#689
| License       | MIT

### CODE REVIEW

**Comment by UlrichKu** [comment]
Can you document this as a necessary step in UPGRADE-7.1.md

**Comment by UlrichKu** [comment]
And there are a number of other tasks in the ticket:

Change for _all_ cronjobs.

Change type annoations.

Do an integrity check in an update.

File: UPGRADE-7.1.md [hunk 1]
```diff
@@ -68,6 +68,24 @@ All service classes that are loaded by `ServiceLocator::get()` must now be `publ
 # Cleanup
 # Informational
 # Changed Features
+## Cronjobs
```
Comment (UlrichKu) [inline UPGRADE-7.1.md:71]: Extending TdbCmsCronjobs (instead of TCMSCronJob) must also be done.

File: UPGRADE-7.1.md [hunk 2]
```diff
@@ -68,6 +68,24 @@ All service classes that are loaded by `ServiceLocator::get()` must now be `publ
 # Cleanup
```
Comment (UlrichKu) [inline UPGRADE-7.1.md:68]: Cronjobs node should be here after Make Legacy Services Public.

And probably be called something like: Correct Cronjob Hierarchy

File: src/CoreBundle/Tests/CronJob/fixtures/CronJobThatExtendsTCMSCronJob.php [hunk 3]
```diff
@@ -6,8 +6,10 @@
 
 class CronJobThatExtendsTCMSCronJob extends TCMSCronJob
```
Comment (UlrichKu) [inline src/CoreBundle/Tests/CronJob/fixtures/CronJobThatExtendsTCMSCronJob.php:7]: This is what must be changed: not extending TCMSCronJob

File: src/CmsResultCacheBundle/objects/CronJobs/TCMSCronJob_PkgCmsResultCache_GarbageCollector.class.php [hunk 4]
```diff
@@ -14,6 +14,12 @@
 /**/
 class TCMSCronJob_PkgCmsResultCache_GarbageCollector extends TCMSCronJob
 {
+
+    public function __construct()
```
Comment (UlrichKu) [inline src/CmsResultCacheBundle/objects/CronJobs/TCMSCronJob_PkgCmsResultCache_GarbageCollector.class.php:18]: This is not necessary.

File: UPGRADE-7.1.md (hunk 1 above)
Comment (kzorluoglu) [reply UPGRADE-7.1.md:71]: updated.

File: UPGRADE-7.1.md (hunk 2 above)
Comment (kzorluoglu) [reply UPGRADE-7.1.md:68]: alright, position changed.

File: src/CmsResultCacheBundle/objects/CronJobs/TCMSCronJob_PkgCmsResultCache_GarbageCollector.class.php (hunk 4 above)
Comment (kzorluoglu) [reply src/CmsResultCacheBundle/objects/CronJobs/TCMSCronJob_PkgCmsResultCache_GarbageCollector.class.php:18]: removed.

File: src/CoreBundle/Tests/CronJob/fixtures/CronJobThatExtendsTCMSCronJob.php (hunk 3 above)
Comment (kzorluoglu) [reply src/CoreBundle/Tests/CronJob/fixtures/CronJobThatExtendsTCMSCronJob.php]: resolved.

File: UPGRADE-7.1.md [hunk 5]
```diff
@@ -65,6 +65,24 @@ as service.
 
 All service classes that are loaded by `ServiceLocator::get()` must now be `public="true"` in their service definitions.
 
+## Correct Cronjob Hierarchy
+
+All Cron Jobs (instead of TCMSCronJob or TdbCmsCronjobs) should now call the superclass constructor.
```
Comment (UlrichKu) [inline UPGRADE-7.1.md:70]: "

Cronjob classes need to extend the class TdbCmsCronjobs. Currently the often used class TCMSCronJob does not have all the necessary fields.

So searching for "TCMSCronJob" in the project code should give you a complete list for this task.

Furthermore every cronjob should only calls the parent constructor like so...

File: src/CmsChangeLogBundle/src/esono/pkgcmschangelog/CronJob/TCMSCronJob_ArchiveChangeLog.class.php [hunk 6]
```diff
@@ -12,8 +12,14 @@
 /**
  * Cronjob for archiving changelog data.
 /**/
-class TCMSCronJob_ArchiveChangeLog extends TCMSCronJob
+class TCMSCronJob_ArchiveChangeLog extends TdbCmsCronjobs
 {
+
+    public function __construct()
```
Comment (UlrichKu) [inline src/CmsChangeLogBundle/src/esono/pkgcmschangelog/CronJob/TCMSCronJob_ArchiveChangeLog.class.php:18]: Empty constructors are not necessary.

**Comment by UlrichKu** [comment]
These points are still open:

Change type annoations.

Do an integrity check in an update.

File: src/CmsChangeLogBundle/src/esono/pkgcmschangelog/CronJob/TCMSCronJob_ArchiveChangeLog.class.php (hunk 6 above)
Comment (kzorluoglu) [reply src/CmsChangeLogBundle/src/esono/pkgcmschangelog/CronJob/TCMSCronJob_ArchiveChangeLog.class.php]: removed.

File: UPGRADE-7.1.md [hunk 7]
```diff
@@ -65,6 +65,42 @@ as service.
...
+    {
+        parent::TdbCmsCronjobs();
+    }
+}
+```
+Should Be:
+```
+class ...Cronjob extends TdbCmsCronjobs {
+
+ public function __construct()
```
Comment (UlrichKu) [inline UPGRADE-7.1.md:98]: (Empty constructors are not needed.)

So proably we should make clear that there is (other) code in the constructor.

File: src/AtomicLockBundle/Cronjobs/TCMSCronjob_ClearAtomicLocks.class.php [hunk 8]
```diff
@@ -9,8 +9,14 @@
  * file that was distributed with this source code.
  */
 
-class TCMSCronjob_ClearAtomicLocks extends TCMSCronJob
+class TCMSCronjob_ClearAtomicLocks extends TdbCmsCronjobs
 {
+
+    public function __construct()
```
Comment (UlrichKu) [inline src/AtomicLockBundle/Cronjobs/TCMSCronjob_ClearAtomicLocks.class.php:15]: Empty constructors are not needed. (12 other occurrances here)

**Comment by UlrichKu** [comment]
These points are still open:

Change type annotations.

Do an integrity check in an update.

File: UPGRADE-7.1.md (hunk 7 above)
Comment (kzorluoglu) [reply UPGRADE-7.1.md]: constructor example updated with demo parameters.

File: src/AtomicLockBundle/Cronjobs/TCMSCronjob_ClearAtomicLocks.class.php (hunk 8 above)
Comment (kzorluoglu) [reply src/AtomicLockBundle/Cronjobs/TCMSCronjob_ClearAtomicLocks.class.php]: all removed.

**Pull request merged**